api/domain/ordersDomain.js

```diff
@@ -0,0 +1,168 @@
+const orderRepo = require('@repos/orderRepository')
+const { Verification, Resources } = require('@verify/verification')
+const { IdNotFound, SequelizeError } = require('@helpers/errors')
+const { OrderHelpers } = require('@repos/orderHelpers/orderHelpers')
+
+exports.create = async (orderData) => {
+    try {
+        await Verification.IdExists(Resources.Customers, orderData.customerId)
+
+        for (let n = 0; n < orderData.wantedProducts.length; n++) 
+            await Verification.IdExists(Resources.Products, orderData.wantedProducts[n].id)
+    
+        const order = await orderRepo.createOrder(orderData)
+
+        return this.addProducts(order, orderData.wantedProducts) 
+    } catch (error) {
+        throw new IdNotFound(error.message + ' && Order has not been placed!')
+    }
+}
+
+exports.update = async (orderId, updatedOrderData) => {
+    try {
+        await Verification.IdExists(Resources.Orders, orderId)
+        const updatedProducts = updatedOrderData.updatedProducts
+        const order = await orderRepo.getById(orderId)
+
+        await this.updateProducts(order, updatedProducts)
+
+        await orderRepo.updateOrder(orderId, updatedOrderData)
+        const updatedOrder = await orderRepo.getById(orderId)
+
+        if (order.currency !== updatedOrderData.currency)
+            await this.changeCurrency(order, updatedOrder)
+        
+        return orderRepo.updateOrder(orderId, updatedOrderData)
+    } catch (error) {
+        if (error.name.includes("Sequelize")) {
+            throw new SequelizeError('Field cannot be null.')
+        }
+        throw new IdNotFound(error.message + ' && None order was updated.')
+    }
+}
+
+exports.delete = async (orderId) => {
+    try {
+        await Verification.IdExists(Resources.Orders, orderId)
+        
+        return orderRepo.deleteOrder(orderId)
+    } catch (error) {
+        throw new IdNotFound(error.message + ' && No order was deleted.')
+    }
+}
+
+exports.getAll = () => {
+    try {
+        return orderRepo.getAll()
+    } catch (error) {
+        throw new Error('No orders were found.')
+    }
+}
+
+exports.getById = async (orderId) => {
+    try {
+        await Verification.IdExists(Resources.Orders, orderId)
+
+        return orderRepo.getById(orderId)
+    } catch (error) {
+        throw new IdNotFound(error.message)
+    }
+}
+
+exports.addProducts = async (order, products) => {
+    try {
+        for (let n = 0; n < products.length; n++) {
+            const productsInOrder = await order.getProducts({ where: { id: products[n].id } })
```

Comment:
maybe you could extract this part of code from line 75 to 81 to a separate method with descriptive name and call it in here because it's hard to figure out what's going on in here